syntax = "proto3";

package book;

import "protos/author.proto";

message Book {
    uint32 id = 1;
    string title = 2;
    uint32 author_id = 3;
}

message CreateBookRequest {
    string title = 1;
    uint32 author_id = 2;
}

message RetrieveBookRequest {
    uint32 id = 1;
}

message UpdateBookRequest {
    uint32 id = 1;
    string title = 2;
    uint32 author_id = 3;
}

message DeleteBookRequest {
    uint32 id = 1;
    string title = 2;
    uint32 author_id = 3;
}


message SingleBookResponse {
    uint32 id = 1;
    string title = 2;
    author.Author author = 3;
}


service BookService {
    rpc CreateBook(CreateBookRequest) returns (SingleBookResponse);
    rpc RetrieveBook(RetrieveBookRequest) returns (SingleBookResponse);
    rpc UpdateBook(UpdateBookRequest) returns (SingleBookResponse);
    rpc DeleteBook(DeleteBookRequest) returns (SingleBookResponse);
}
